pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1531,6 +1535,12 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (string, error) {
 	volSize := int64(volumeOptions.CapacityGB)
 	request.Size = &volSize
 	request.VolumeType = &createType
+	request.Encrypted = &volumeOptions.Encrypted
+	request.KmsKeyId = &volumeOptions.KmsKeyId
+	if len(*request.KmsKeyId) > 0 {
```

Comment:
Nit: I would probably go to the source here - i.e. `if len(volumeOptions.KmsKeyId) > 0`.  (Not sure why we do that rather than check != "", but it is our style).